src/Microsoft.Health.Dicom.Core/Features/Query/Model/SeriesResult.cs

```diff
@@ -41,4 +42,14 @@ public DicomDataset DicomDataset
             return _dicomDataset;
         }
     }
+
+    public static readonly ImmutableHashSet<DicomTag> AvailableTags = ImmutableHashSet.Create<DicomTag>
```

Comment:
Not exactly. things in the QueryResponseBuilder are the contracts to customer response. This class just represent the tags that are indexed and avaliable from the Index/SQLServer

Reply:
Yes - my point is that we already have classes that define those sets: `QueryLimit` for tags that are indexed by default, and `QueryResponseBuilder` for the API response contract. Why not use the existing definitions of available tags in `QueryLimit`?

Reply:
Not the same. Separation of concerns. 
QueryLimit: Contains things we support during filtering.
QueryResponseBuilder: Contains the tags that are returned by default.
StudyResult/SeriesResult: Contains the tags that are available through SQL

All these can independently change. What is available in SQL should not be tied to what the default tags are returned

Reply:
Ok - I would like to make sure I understand the distinction you're making here between 
> QueryLimit: Contains things we support during filtering.

and 

>StudyResult/SeriesResult: Contains the tags that are available through SQL

To me, those sound like the same thing - if we add a tag to be supported during filtering, it's by definition present in the SQL table, and vice versa.

Are you saying that the set of tags in StudyResult/SeriesResult model what is returned by the GetStudyResult / GetSeriesResult sprocs, and we want that to be able to independently change from what tags are stored in the tables?

Reply:
Yes. There are computed columns that I put in StudyResultComputedTags that together with CoreFilterTags that form the Result. The StudyResult class just represents the SQL columns, since the property is just in the class it will be easy to add it there.